### DISCUSSION TITLE
Installing xfig

### DISCUSSION THREAD

**Discussion by adamny** [Homebrew/discussions#5037] [Getting started]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: b3751bca8c4a3c76107d5aa3b75b81db93cf4bb6
Last commit: 2 days ago
Core tap JSON: 09 Jan 16:37 UTC
Core cask tap JSON: 09 Jan 16:37 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.yxbYRS4AZj/org.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.0-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I executed: "brew install xfig". No error messages. The last three lines of output where:

==> Installing xfig
==> Pouring xfig--3.2.9.arm64_sonoma.bottle.tar.gz
🍺  /opt/homebrew/Cellar/xfig/3.2.9: 3,005 files, 14.2MB
==> Running `brew cleanup xfig`...

I see xfig in /opt/homebrew/bin
But when I execute it in mac terminal I get: 

> Warning: locale not supported by Xlib, locale set to C
> Warning: X locale modifiers not supported, using default
> Warning: Cannot convert string "-*-times-medium-r-normal--16-*-*-*-*-*-*-*,-*-*-medium-r-normal--16-*-*-*-*-*-*-*,*--16-*" to type FontSet
> Warning: Unable to load any usable fontset
> zsh: segmentation fault  xfig

I get the same error when I execute it in xterm under XQuartz. How can I fix it?

**Comment by fahasch** [comment]
Same problem here... It turns out that the segmentation fault only happens on apple silicon. On intel processors everything is fine. 
This is the backtrace from my process
```
    frame #0: 0x00000001008b5500 libX11.6.dylib`XmbTextEscapement
    frame #1: 0x0000000100687424 libXaw3d.8.dylib`GetDefaultSize + 92
    frame #2: 0x000000010072b604 libXt.6.dylib`CallInitialize + 176
    frame #3: 0x000000010072a940 libXt.6.dylib`xtCreate + 524
    frame #4: 0x000000010072a64c libXt.6.dylib`_XtCreateWidget + 408
    frame #5: 0x000000010006140c xfig`rebuild_file_menu + 240
    frame #6: 0x0000000100061760 xfig`create_menu_item + 296
    frame #7: 0x00000001000610d8 xfig`init_main_menus + 480
    frame #8: 0x00000001000a2968 xfig`main + 3144
    frame #9: 0x00000001848610e0 dyld`start + 2360
```
So it fails while calling the X11 library.

**Reply by peterlittig** [reply]
Same problem here. Any word on a potential fix?

**Reply by fahasch** [reply]
I reported the issue [here](https://github.com/Homebrew/homebrew-core/issues/160515). Have not heard back.

**Comment by kernsb** [comment]
This can be worked around by disabling "international" font support.

Solution:

 1. In a plain text editor, open the file at
    **/opt/homebrew/etc/X11/app-defaults/Fig**
    

 2. Locate the line that contains the key of "Fig.international" and set it to "False"

    ```
    Fig.international: False
    ```

 3. In a plain text editor, open the file at
    **/opt/homebrew/Cellar/xfig/3.2.9/.bottle/etc/X11/app-defaults/Fig**
    
    *Note: The version number in this path was at the time of this writing and may need to be adjusted to your installed version.*

 4. Locate the line that contains the key of "Fig.international" and set it to "False"

    ```
    Fig.international: False
    ```

 5. Launch Xfig as usual

**Reply by SMillerDev** [reply]
Really only the first one should be needed. Can you make a pull request that does this on Apple silicon? And/or report this issue upstream?

**Reply by fahasch** [reply]
It has already been [reported](https://sourceforge.net/p/mcj/tickets/174/). The recommendation until it is fixed is to change the configuration file.

**Comment by adamny** [comment]
Many thanks for this fix.

The only shortcoming of it is that xfig has Roman fonts only now. Would
anyone have an idea how to install Greek letters? They are often used for
scientific drawings.

Being able to run it is definitely a progress.

-- Adam

**Reply by fahasch** [reply]
It seems that the problem is mostly with the Symbol font. Selecting other fonts works fine. Also it seems to be a problem with the rendering (note that the number 1-9 in Symbol work fine). Also when converting the fig-file to pdf, the letters appear as expected. I would suggest to report this bug upstream.
